### DISCUSSION TITLE
SRP server unable to handle big TXT records

### DISCUSSION THREAD

**Discussion by RahulRGaikwad** [openthread/openthread#9834] [Q&A]
Hello everyone, 

Currently trying to use SRP client to register information, to a SRP server. 

The SRP server in question, is an ESP32-C6 flashed with ot_br example code. 

First I tried using SRP client buffers to allocate service entries. This was not possible as each service entry object, can only contain one TXT entry object. 

So I manually created a SRP service object, and was able to assign multiple TXT entries. 

But upon registering the SRP server crashes, when the total encoded buffer of the TXT entries crosses 64 bytes.

The crash log:
![tmp_95ad91e0-9340-4cb6-a84e-16aef9d658b3.png](https://github.com/openthread/openthread/assets/74486473/bbd3a628-b42d-4676-aa0c-e4219cc0d06c)

Looks like it's the SRP handler registered by ot_br, that's crashing. 

Any help on how to resolve this issue, will be helpful

**Comment by chshu** [comment]
@RahulRGaikwad The TXT buffer size is limited to 64 in openthread by default: ([OPENTHREAD_CONFIG_SRP_CLIENT_BUFFERS_TXT_BUFFER_SIZE](https://github.com/openthread/openthread/blob/main/src/core/config/srp_client.h#L412))

Each service entry could contain more than one Txt entry, you can register it with the ot cli command:
```
srp client service add thread-service _test._tcp,_sub1,_sub2 12345 1 1 0778797a3d58595a0778797a3d58595a0778797a3d58595a0778797a3d58595a
```

If bigger TXT size is required, you can increase the OPENTHREAD_CONFIG_SRP_CLIENT_BUFFERS_TXT_BUFFER_SIZE config. I tried to increase it to 80, and register a service with 80 bytes TXT:
```
srp client service add thread-service _test._tcp,_sub1,_sub2 12345 1 1 0778797a3d58595a0778797a3d58595a0778797a3d58595a0778797a3d58595a0778797a3d58595a0778797a3d58595a0778797a3d58595a0778797a3d58595a0778797a3d58595a0778797a3d58595a
```

Then the ESP TBR works well:
```
> srp server service

thread-service._test._tcp.default.service.arpa.
    deleted: false
    subtypes: _sub1,_sub2
    port: 12345
    priority: 1
    weight: 1
    ttl: 7200
    lease: 7200
    key-lease: 680400
    TXT: [xyz=58595a, xyz=58595a, xyz=58595a, xyz=58595a, xyz=58595a, xyz=58595a, xyz=58595a, xyz=58595a, xyz=58595a, xyz=58595a]
    host: thread-device.default.service.arpa.
    addresses: [fd6f:ef12:5ae:1:ba4f:578f:fd63:bc90]
thread-ser._threads._tcp.default.service.arpa.
    deleted: true
Done
```
It can also be discoved by avahi-browse:
```
➜  basic_thread_border_router git:(main) avahi-browse -r _test._tcp
+ enp1s0 IPv6 thread-service                                _test._tcp           local
+ enp1s0 IPv4 thread-service                                _test._tcp           local
= enp1s0 IPv6 thread-service                                _test._tcp           local
   hostname = [thread-device.local]
   address = [fd6f:ef12:5ae:1:ba4f:578f:fd63:bc90]
   port = [12345]
   txt = ["xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ"]
= enp1s0 IPv4 thread-service                                _test._tcp           local
   hostname = [thread-device.local]
   address = [fd6f:ef12:5ae:1:ba4f:578f:fd63:bc90]
   port = [12345]
   txt = ["xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ" "xyz=XYZ"]
```

Generally it works as expected. Could you please share the steps to reproduce the crash issue?
